### ISSUE TITLE
Ollama Fails to Install on Linux

### ISSUE THREAD

**Issue by brian-at-pieces** [pieces-app/support#537]
### Software

Desktop Application

### Operating System / Platform

Linux

### Your Pieces OS Version

11.0.0

### Early Access Program

- [ ] Yes, this is related to an Early Access Program feature.

### Kindly describe the bug and include as much detail as possible on what you were doing so we can reproduce the bug.

User reports immediate "Ollama failed to install" diaglog upon clicking install Ollama install button in Pieces desktop app

**Comment by brian-at-pieces** [comment]
@ekuznetsov139  I believe the issue is that one of our packages is trying to using the `curl` program which is not guaranteed to be installed. We'll be patching this, but if you `apt install curl` you should be able to complete the download

**Comment by ekuznetsov139** [comment]
I do have curl, it is a standard part of ubuntu. 

I have downloaded and installed ollama manually, I can launch it from a terminal and it runs in the background, but that had no effect on Pieces, which still wants to install it and still immediately prints "Ollama failed to install" .

**Comment by ekuznetsov139** [comment]
Ok, I think I see the problem. It's not that _I_ don't have curl. The problem is that pieces-os is installed as a snap, and it runs in its own sandboxed environment from which curl is missing. You should add curl to the list of dependency packages somewhere in the snap definition file.

For the same reason, installing ollama globally would have no effect, because pieces-os can't launch or even see any executables outside its sandbox.

**Comment by brian-at-pieces** [comment]
Yeah you're absolutely right. A little miscommunication internally about whether our package has strict or class confinement. We're working on a patch that allows us to use ollama from a sandbox. Sorry again for the inconvenience here. We'll have a solution out ASAP

**Comment by ekuznetsov139** [comment]
The issue is still present in the latest version available via snap on Ubuntu as of today.

**Comment by brian-at-pieces** [comment]
Sorry about the delay on this - we're finishing up some critical infrastructure updates that unfortunately took priority. I have the fix on a branch and will hopefully get it in next week. Thanks for the patience

**Comment by cwoolfo1** [comment]
Has there been any progress on fixing this?

**Comment by cwoolfo1** [comment]
Is there a way to get local llm models to download in the meantime?

**Comment by brian-at-pieces** [comment]
Unfortunately this issue impacts inference as well as downloading, so manually downloading the models will not resolve it. Working on this as we speak

**Comment by umeshwayakole27** [comment]
Please try to solve this issue early i am waiting for it to solve from very long time because I don't want to give my whole computer data to the cloud LLMs which could access anything

**Comment by JeesusKrisostoomus** [comment]
Any update on this?

**Comment by senyor** [comment]
Is `curl` missing?
I've got the following error in debug mode for Ollama installation in pieces-os
```
flutter: [D]  2025-03-19T14:19:37.455178  0:00:00.173781 POST    [200] /ollama/install: port:39301
ERROR - 2025-03-19 14:19:37.652748
Asynchronous error
Exception: Error during Linux OllamaOperation.install: ProcessException: No such file or directory
  Command: curl --version
package:isomorphic_server/ollama/ollama_facade.dart 96  OllamaFacade.installOllama
```

**Comment by phbrgnomo** [comment]
Here to bump this. Just installed on ubuntu, ollama running good, but pieces os can't find it.
